### PR TITLE
Connector type list

### PR DESCRIPTION
**Pull request by jackdelahunt**
### Verification Steps

1. `rhoas connector type list`
2. `rhoas connector type list --output=yaml`
3. `rhoas connector type list --limit=5`
4. `rhoas connector type list --search=Amazon%`
5. `rhoas connector type list --limit=15 --page=2`

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### CODE REVIEW

File: pkg/cmd/connector/connector_type/list/list.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+	types, httpRes, err := request.Execute()
+
+	if httpRes != nil {
+		defer httpRes.Body.Close()
+	}
+
+	if err != nil {
+		switch httpRes.StatusCode {
+		case http.StatusUnauthorized:
```
Comment (wtrocki) [inline pkg/cmd/connector/connector_type/list/list.go:96]: This is antipattern. We should not use status codes for validation. We should use `error codes`. 
See kafka create for example

**Review by wtrocki — APPROVED** [review]
Works. Minor stuff reported

File: pkg/cmd/connector/connector.go [hunk 2]
```diff
@@ -36,6 +37,8 @@ func NewConnectorsCommand(f *factory.Factory) *cobra.Command {
 		use.NewUseCommand(f),
 		start.NewStartCommand(f),
 		stop.NewStopCommand(f),
+		connector_type.NewTypeCommand(f),
```
[the remark this answers is not in the document]
Comment (jackdelahunt) [reply pkg/cmd/connector/connector.go:40]: This is due to `type` being a reserved word in go. I just went with connector_type instead.

File: pkg/cmd/connector/connector_type/list/list.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+			return runUpdateCommand(opts)
+		},
+	}
+
+	flags := connectorcmdutil.NewFlagSet(cmd, f)
+	flags.AddOutput(&opts.outputFormat)
+	flags.IntVar(&opts.limit, "limit", 150, f.Localizer.MustLocalize("connector.type.list.flag.page.description"))
+	flags.StringVar(&opts.search, "search", DefaultSearch, f.Localizer.MustLocalize("connector.type.list.flag.search.description"))
+	flags.IntVar(&opts.page, "page", 1, f.Localizer.MustLocalize("connector.type.list.flag.page.description"))
```
[the remark this answers is not in the document]
Comment (wtrocki) [reply pkg/cmd/connector/connector_type/list/list.go:63]: Yes. That is common pattern

**Pull request merged**